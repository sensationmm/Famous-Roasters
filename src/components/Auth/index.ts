import { Authenticator } from './Authenticator'
import { AuthFormEmail } from './AuthFormEmail'
import { AuthFormItemInput } from './AuthFormItemInput'
import { AuthFormPassword } from './AuthFormPassword'
import { AuthSignIn } from './SignIn'

export { Authenticator as Authenticator }
export { AuthFormEmail as AuthFormEmail }
export { AuthFormItemInput as AuthFormItemInput }
export { AuthFormPassword as AuthFormPassword }
export { AuthSignIn as AuthSignIn }
