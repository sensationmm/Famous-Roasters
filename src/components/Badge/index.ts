import { Badge } from './Badge'

export { Badge as Badge }
