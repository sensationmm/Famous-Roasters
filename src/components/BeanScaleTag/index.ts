import { BeanScaleTag } from './BeanScaleTag'

export { BeanScaleTag as BeanScaleTag }
