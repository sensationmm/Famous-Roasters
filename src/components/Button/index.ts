import { Button, Color, Emphasis, Size } from './Button'

export { Button as Button }
export { Color as ButtonColor }
export { Emphasis as ButtonEmphasis }
export { Size as ButtonSize }
