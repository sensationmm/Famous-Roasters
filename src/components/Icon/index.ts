import { Icon, IconName, IconSize } from './Icon'

export { Icon as Icon }
export { IconName as IconName }
export { IconSize as IconSize }
