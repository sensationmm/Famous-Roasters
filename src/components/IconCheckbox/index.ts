import { IconCheckbox } from './IconCheckbox'

export { IconCheckbox as IconCheckbox }
