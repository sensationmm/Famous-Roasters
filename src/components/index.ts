import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
