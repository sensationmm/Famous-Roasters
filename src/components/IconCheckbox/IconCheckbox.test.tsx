import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { IconName } from 'src/components/Icon'

import { IconCheckbox } from '.'

global.alert = jest.fn()

describe('ImageCheckbox component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <IconCheckbox name="test" iconName={IconName.Espresso} text="Dieser Kaffee ist lekka" />,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for selected', async () => {
    const { container } = render(
      <IconCheckbox name="test" iconName={IconName.Espresso} text="Dieser Kaffee ist lekka" selected={true} />,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can toggle the state by clicking', async () => {
    render(
      <IconCheckbox
        name="test"
        iconName={IconName.Espresso}
        text="Dieser Kaffee ist lekka"
        toggleSelected={() => alert('works')}
      />,
    )
    const iconCheckbox = await screen.findByTestId('icon-checkbox')
    expect(iconCheckbox).toBeInTheDocument()
    fireEvent.click(iconCheckbox)
    fireEvent.click(iconCheckbox)
  })
})
