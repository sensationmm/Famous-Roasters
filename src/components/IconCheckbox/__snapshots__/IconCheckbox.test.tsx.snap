// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCheckbox component Renders correctly 1`] = `
<div>
  <button
    class="flex w-full flex-row pl-6 py-4 items-center cursor-pointer"
    data-testid="icon-checkbox"
  >
    <div>
      <svg
        height="24"
        viewBox="0 0 16 16"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M8,16c-4.4,0-8-3.6-8-8s3.6-8,8-8s8,3.6,8,8S12.4,16,8,16z M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z"
          fill="#D8D8D8"
          fill-rule="evenodd"
          stroke="#D8D8D8"
          stroke-width="0.1"
        />
      </svg>
    </div>
    <div
      class="flex w-16 h-16 ml-4 mr-4 rounded-full bg-brand-grey-whisper items-center justify-center shrink-0"
    >
      <svg
        height="32"
        viewBox="0 0 30 32"
        width="32"
      >
        <path
          clip-rule="evenodd"
          d="M27.31 0H11.42a2.494 2.494 0 0 0-2.482 2.5v6c0 1.379 1.114 2.5 2.483 2.5h.496v.5H9.842a1.499 1.499 0 0 0-1.19-.985C8.504 10.494 5.024 10 1.985 10A1.995 1.995 0 0 0 0 12c0 1.103.89 2 1.986 2 3.039 0 6.519-.494 6.666-.515a1.5 1.5 0 0 0 1.19-.985h2.075v.5c0 .276.223.5.497.5h.727l.846 1.277c.092.14.247.223.413.223h.496v.5c0 .276.223.5.497.5a.499.499 0 0 0 .497-.5V15h.496c.166 0 .321-.084.413-.223l.846-1.277h.727a.499.499 0 0 0 .497-.5v-2h2.483a.5.5 0 0 1 .496.5v12a.5.5 0 0 1-.496.5h-2.495c.314-.419.509-.935.509-1.5V18c0-.276-.223-.5-.497-.5h-6.952a.499.499 0 0 0-.496.5v.092a1.464 1.464 0 0 0-.497-.092h-.993c-.821 0-1.49.673-1.49 1.5v2c0 .827.669 1.5 1.49 1.5h.993c.19 0 .37-.04.536-.104.066.41.23.785.47 1.104h-1.502a2.494 2.494 0 0 0-2.483 2.5v3c0 1.379 1.113 2.5 2.482 2.5H27.31a2.494 2.494 0 0 0 2.483-2.5v-27c0-1.379-1.114-2.5-2.483-2.5ZM8.513 12.495c-.144.02-3.557.505-6.526.505a.998.998 0 0 1-.993-1c0-.552.445-1 .993-1 2.97 0 6.382.484 6.526.505a.501.501 0 0 1 0 .99ZM11.42 1h15.89c.82 0 1.49.673 1.49 1.5H9.93c0-.827.668-1.5 1.49-1.5Zm4.7 13h-1.455l-.332-.5h2.117l-.33.5Zm1.755-1.5H12.91V11h4.966v1.5ZM10.924 22h-.993a.5.5 0 0 1-.497-.5v-2a.5.5 0 0 1 .497-.5h.993a.5.5 0 0 1 .497.5v2a.5.5 0 0 1-.497.5Zm1.49.5v-4h5.958v4c0 .827-.668 1.5-1.49 1.5h-2.979c-.82 0-1.49-.673-1.49-1.5ZM10.428 25h10.924c.821 0 1.49-.673 1.49-1.5v-12c0-.827-.669-1.5-1.49-1.5H11.42c-.822 0-1.49-.673-1.49-1.5v-5h18.87v25H8.937v-2c0-.827.668-1.5 1.49-1.5Zm16.882 6H10.428c-.822 0-1.49-.673-1.49-1.5H28.8c0 .827-.668 1.5-1.49 1.5Z"
          fill="#212121"
          fill-rule="evenodd"
          stroke="#212121"
          stroke-width="0.25"
        />
      </svg>
    </div>
    <div
      class="flex grow text-left text-base leading-6 font-semibold"
    >
      Dieser Kaffee ist lekka
    </div>
  </button>
</div>
`;

exports[`ImageCheckbox component Renders correctly for selected 1`] = `
<div>
  <button
    class="flex w-full flex-row pl-6 py-4 items-center cursor-pointer"
    data-testid="icon-checkbox"
  >
    <div>
      <svg
        height="24"
        viewBox="0 0 16 16"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M8,0C3.6,0,0,3.6,0,8s3.6,8,8,8s8-3.6,8-8S12.4,0,8,0z M8,11c-1.7,0-3-1.3-3-3s1.3-3,3-3s3,1.3,3,3S9.7,11,8,11z"
          fill="#212121"
          fill-rule="evenodd"
          stroke="#212121"
          stroke-width="0.1"
        />
      </svg>
    </div>
    <div
      class="flex w-16 h-16 ml-4 mr-4 rounded-full bg-brand-grey-whisper items-center justify-center shrink-0"
    >
      <svg
        height="32"
        viewBox="0 0 30 32"
        width="32"
      >
        <path
          clip-rule="evenodd"
          d="M27.31 0H11.42a2.494 2.494 0 0 0-2.482 2.5v6c0 1.379 1.114 2.5 2.483 2.5h.496v.5H9.842a1.499 1.499 0 0 0-1.19-.985C8.504 10.494 5.024 10 1.985 10A1.995 1.995 0 0 0 0 12c0 1.103.89 2 1.986 2 3.039 0 6.519-.494 6.666-.515a1.5 1.5 0 0 0 1.19-.985h2.075v.5c0 .276.223.5.497.5h.727l.846 1.277c.092.14.247.223.413.223h.496v.5c0 .276.223.5.497.5a.499.499 0 0 0 .497-.5V15h.496c.166 0 .321-.084.413-.223l.846-1.277h.727a.499.499 0 0 0 .497-.5v-2h2.483a.5.5 0 0 1 .496.5v12a.5.5 0 0 1-.496.5h-2.495c.314-.419.509-.935.509-1.5V18c0-.276-.223-.5-.497-.5h-6.952a.499.499 0 0 0-.496.5v.092a1.464 1.464 0 0 0-.497-.092h-.993c-.821 0-1.49.673-1.49 1.5v2c0 .827.669 1.5 1.49 1.5h.993c.19 0 .37-.04.536-.104.066.41.23.785.47 1.104h-1.502a2.494 2.494 0 0 0-2.483 2.5v3c0 1.379 1.113 2.5 2.482 2.5H27.31a2.494 2.494 0 0 0 2.483-2.5v-27c0-1.379-1.114-2.5-2.483-2.5ZM8.513 12.495c-.144.02-3.557.505-6.526.505a.998.998 0 0 1-.993-1c0-.552.445-1 .993-1 2.97 0 6.382.484 6.526.505a.501.501 0 0 1 0 .99ZM11.42 1h15.89c.82 0 1.49.673 1.49 1.5H9.93c0-.827.668-1.5 1.49-1.5Zm4.7 13h-1.455l-.332-.5h2.117l-.33.5Zm1.755-1.5H12.91V11h4.966v1.5ZM10.924 22h-.993a.5.5 0 0 1-.497-.5v-2a.5.5 0 0 1 .497-.5h.993a.5.5 0 0 1 .497.5v2a.5.5 0 0 1-.497.5Zm1.49.5v-4h5.958v4c0 .827-.668 1.5-1.49 1.5h-2.979c-.82 0-1.49-.673-1.49-1.5ZM10.428 25h10.924c.821 0 1.49-.673 1.49-1.5v-12c0-.827-.669-1.5-1.49-1.5H11.42c-.822 0-1.49-.673-1.49-1.5v-5h18.87v25H8.937v-2c0-.827.668-1.5 1.49-1.5Zm16.882 6H10.428c-.822 0-1.49-.673-1.49-1.5H28.8c0 .827-.668 1.5-1.49 1.5Z"
          fill="#212121"
          fill-rule="evenodd"
          stroke="#212121"
          stroke-width="0.25"
        />
      </svg>
    </div>
    <div
      class="flex grow text-left text-base leading-6 font-semibold"
    >
      Dieser Kaffee ist lekka
    </div>
  </button>
</div>
`;
